connectionclass/src/main/java/com/facebook/network/connectionclass/DeviceBandwidthSampler.java

```diff
@@ -100,38 +110,28 @@ public void handleMessage(Message msg) {
           addSample();
           sendEmptyMessageDelayed(MSG_START, SAMPLE_TIME);
           break;
-        case MSG_STOP:
-          addFinalSample();
-          removeMessages(MSG_START);
-          break;
         default:
           throw new IllegalArgumentException("Unknown what=" + msg.what);
       }
     }
+  }
 
-    /**
-     * Method for polling for the change in total bytes since last update and
-     * adding it to the BandwidthManager.
-     */
-    private void addSample() {
-      long byteDiff = QTagParser.getInstance().parseDataUsageForUidAndTag(Process.myUid());
+  /**
+   * Method for polling for the change in total bytes since last update and
+   * adding it to the BandwidthManager.
+   */
+  protected void addSample() {
+    long newBytes = TrafficStats.getTotalRxBytes();
```

Comment:
Good catch.  Didn't even realize this is exactly what we're doing now and it's also written in C by the platform.